### DISCUSSION TITLE
Need help detecting what document is open in a custom editor

### DISCUSSION THREAD

**Discussion by david-pa** [microsoft/vscode-discussions#482] [Extension Development QnA] [answered]
I'm currently working on a proof of concept and have been modifying the [CatScratch/PawDraw VSCode extension example](https://github.com/microsoft/vscode-extension-samples/tree/main/custom-editor-sample) for custom (webview) editors. I have a button currently in the `editor/title` contribution point that is meant to fire off a command that closes the active editor and opens up the custom CatScratch editor or the default text editor depending on which is open (CatScratch --> text edtior, text editor --> CatScratch).

Right now I'm having trouble identifying a good way of grabbing the current document that is open in a custom editor. Given that it's not a text editor, you can't use `vscode.window.activeTextEditor?.document` to grab the document, as there's no text editor active. Is there a more straightforward way to get either the document URI or some information that can allow me to grab the open document when a custom editor is the active editor? I just need to be able to grab a document reference that can be used with `vscode.window.showTextDocument(document, vscode.ViewColumn.Active, false)`.

**Comment by ArturoDent** [comment, marked as answer]
Look at the `Tab.input` property - it should have the `uri` only for a text editor and `viewType` only for a webiew.  I think you will have to get to the current editor through `Tabgroups.activeTabGroup.activeTab`.

**Reply by david-pa** [reply]
The issue is that I'm looking to figure out what document is open/used by a webview, though. Since this custom editor isn't a text editor, I think it'd only list the `viewType`, which I don't think would help me in this case.

**Reply by david-pa** [reply]
I spent some time on this and I understand what you mean now! I'm sorry, it wasn't very clear at first.

In order to get the URI of the document open in the custom editor, I can use `vscode.window.tabGroups.activeTabGroup.activeTab.input`, which will allow me to access the properties `uri` and `viewType`. 
Then I can throw that data into a variable like `activeTabInput` and use `await vscode.workspace.openTextDocument(activeTabInput.uri)` to get a document back.
